### ISSUE TITLE
Issues parsing numeric MailboxIds 

### ISSUE THREAD

**Issue by rudihorn** [stalwartlabs/jmap-server#50]
I am unable to send an email using stalwart, because parsing the following request fails:

```
{
  "using": [
    "urn:ietf:params:jmap:submission",
    "urn:ietf:params:jmap:mail",
    "urn:ietf:params:jmap:core"
  ],
  "methodCalls": [
    [
      "Email/set",
      {
        "accountId": "c",
        "create": {
          "c37ee58b-e224-4799-88e6-1d7484e3b782": {
            "mailboxIds": {
              "9": true
            },
            "subject": "test",
            "from": [
              {
                "name": "Foo",
                "email": "foo@bar.com"
              }
            ],
            "to": [
              {
                "name": null,
                "email": "bar@foo.com"
              }
            ],
            "cc": [],
            "bcc": [],
            "replyTo": [
              {
                "name": null,
                "email": "foo@bar.com"
              }
            ],
            "htmlBody": [
              {
                "partId": "c37ee58b-e224-4799-88e6-1d7484e3b782",
                "type": "text/html"
              }
            ],
            "bodyValues": {
              "c37ee58b-e224-4799-88e6-1d7484e3b782": {
                "value": "<p>test email<br></p>",
                "isEncodingProblem": false,
                "isTruncated": false
              }
            },
            "header:User-Agent:asText": "Team-Mail/0.7.11 Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/113.0"
          }
        }
      },
      "c0"
    ],
    [
      "EmailSubmission/set",
      {
        "accountId": "c",
        "create": {
          "c37ee58b-e224-4799-88e6-1d7484e3b782": {
            "identityId": "a",
            "emailId": "#c37ee58b-e224-4799-88e6-1d7484e3b782",
            "envelope": {
              "mailFrom": {
                "email": "foo@bar.com"
              },
              "rcptTo": [
                {
                  "email": "bar@foo.com"
                }
              ]
            }
          }
        },
        "onSuccessUpdateEmail": {
          "#c37ee58b-e224-4799-88e6-1d7484e3b782": {
            "mailboxIds/d": true,
            "mailboxIds/9": null,
            "keywords/$seen": true,
            "keywords/$draft": null
          }
        }
      },
      "c1"
    ]
  ]
}
```

A bit of hacking around seems to indicate that it is unable to use serde to parse this file. The parser fails with `expected `,` or `]` at line 80 column 27`. From what I can tell it does not like the `mailboxIds/9` part. Changing it to something like `mailboxIds/a` allows it to parse. 

The email was generated with tmail flutter version 0.7.11.

I'm testing the serialization using:

```
mod mytests {
    use crate::api::request::Request;
    use std::io::Read;
    use std::path::Path;
    use std::fs::File;

    #[test]
    fn test() {
        let path = Path::new("email.json");
        let mut request = File::open(path).unwrap();

        let mut text = String::new();
        request.read_to_string(&mut text).unwrap();

        let res = serde_json::from_slice::<Request>(text.as_bytes());
        eprintln!("is ok {}", res.is_ok());
        eprintln!("err {}", res.unwrap_err());

        ()
    }
}
```

**Comment by mdecimus** [comment]
Is mailboxId 9 a valid mailbox id that exists in the server? If the ID cannot be decoded the entire request fails but this is something that has been fixed in the new version that will be released in 1 to 2 months.

**Comment by rudihorn** [comment]
This fails is in the parser, so before the server should be checking the existence of mailboxes. I am also testing this in a unit test, which does not have mailboxes to check.

**Comment by mdecimus** [comment]
Yes, the current parser is based on `serde` and fails if a value cannot be parsed. The new version uses a custom parser and does not have this problem. I have just tested your JSON using the new parser at https://github.com/stalwartlabs/mail-server/blob/main/crates/jmap-proto/src/request/parser.rs and it was able to parse it correctly.

**Comment by rudihorn** [comment]
Sounds good. Looking forward to the next release!

**Comment by mdecimus** [comment]
Hi @rudihorn this issue has now been fixed on Stalwart JMAP v0.3.
